.github/workflows/deploy_fossil_offchain.yml

```diff
@@ -0,0 +1,97 @@
+name: Build and Run Yokai Agent Docker Container
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  AWS_REGION: us-east-1
+  ECR_REPOSITORY: fossil-offchain-processor-images-repository
+  ECS_SERVICE: fossil-offchain-processor-app-ecs
+  ECS_CLUSTER: fossil-offchain-processor-ecs-cluster
+  CONTAINER_NAME: fossil-offchian-app 
+  TASK_NAME: fossil-offchain-processor-app-ecs
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  deploy-agent:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::654654236251:role/terraform-20241003185909187100000001
+          role-session-name: Github
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: "true"
+
+      - name: Check if Docker is installed
+        run: |
+          if ! command -v docker &> /dev/null
+          then
+            sudo apt-get update
+            sudo apt-get install -y docker.io
+            sudo apt-get install -y docker-compose
+          fi
```

Comment:
Pretty sure docker is already available in github actions by default, but if not I prefer to use the actions below:

```
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v3
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
```

This action will properly setup buildx depedencies and multi platform builds as well, so I think its a better alternative.